File: backend/app/schemas/branding.py
"""Тема организации: контракт цветов, которые клиент применяет как есть.

Светофор состояний (`good/caution/risk/low`) в теме отсутствует физически — это
язык продукта, общий для всех организаций (раздел 4.1 дизайн-ТЗ). Переопределить
его нельзя даже случайно: поля под него просто нет.
"""

import uuid
from typing import Annotated

from pydantic import BaseModel, ConfigDict, Field, StringConstraints

#: #RRGGBB или #RRGGBBAA — восьмизначные нужны для полупрозрачных обводок
HexColor = Annotated[str, StringConstraints(pattern=r"^#(?:[0-9a-fA-F]{6}|[0-9a-fA-F]{8})$")]


class ThemeTokens(BaseModel):
    """Цвета темы. Значения по умолчанию — тема продукта (mobile/src/theme/index.ts)."""

    model_config = ConfigDict(extra="forbid")

    # Поверхности
    bg: HexColor = "#0A0D13"
    bg_top: HexColor = "#10161F"
    surface: HexColor = "#131A26"
    surface_2: HexColor = "#1B2432"
    border: HexColor = "#26303F"
    border_bright: HexColor = "#33405280"
    # Текст
    text: HexColor = "#EDF2F9"
    text_muted: HexColor = "#8A99AD"
    # Брендовый слой — то, что меняет организация
    brand: HexColor = "#2D8CFF"
    brand_dark: HexColor = "#1B6FE8"
    #: Бренд как текст и иконки на тёмном фоне: тёмный клубный цвет на нём нечитаем,
    #: поэтому это отдельный, осветлённый оттенок, а не тот же `brand`.
    brand_on: HexColor = "#2D8CFF"
    #: Текст поверх брендовой заливки
    on_brand: HexColor = "#FFFFFF"
    #: Второй клубный цвет; по умолчанию совпадает с основным
    brand_2: HexColor = "#2D8CFF"
    brand_2_on: HexColor = "#2D8CFF"
    # Градиенты: [начало, конец]
    gradient_brand: tuple[HexColor, HexColor] = ("#3D96FF", "#1B6FE8")
    gradient_screen: tuple[HexColor, HexColor] = ("#111927", "#0A0D13")


DEFAULT_TOKENS = ThemeTokens()


class BrandingOut(BaseModel):
    """Ответ клиенту: тема применяется как есть, без ветвлений."""

    #: 0 — тема продукта по умолчанию, организация своей не имеет
    version: int = Field(ge=0)
    org_id: uuid.UUID | None = None
    org_name: str | None = None
    logo_url: str | None = None
    tokens: ThemeTokens

File: mobile/src/api/branding.ts
// Тема организации: сервер — источник правды, клиент применяет её как есть
// (docs/plan-org-branding.md).

import { useQuery } from '@tanstack/react-query';

import { session } from '@/auth/session';
import { type Theme, defaultTheme } from '@/theme';

import { API_URL, api } from './client';

/** Ответ бэкенда: цвета в snake_case, как и во всём остальном API. */
export interface BrandingResponse {
  version: number;
  org_id: string | null;
  org_name: string | null;
  logo_url: string | null;
  tokens: {
    bg: string;
    bg_top: string;
    surface: string;
    surface_2: string;
    border: string;
    border_bright: string;
    text: string;
    text_muted: string;
    brand: string;
    brand_dark: string;
    brand_on: string;
    on_brand: string;
    brand_2: string;
    brand_2_on: string;
    gradient_brand: [string, string];
    gradient_screen: [string, string];
  };
}

/** Цвета приходят с сервера, форма и типографика пока продуктовые. */
export function toTheme(branding: BrandingResponse): Theme {
  const c = branding.tokens;
  return {
    ...defaultTheme,
    bg: c.bg,
    bgTop: c.bg_top,
    surface: c.surface,
    surface2: c.surface_2,
    border: c.border,
    borderBright: c.border_bright,
    text: c.text,
    textMuted: c.text_muted,
    brand: c.brand,
    brandDark: c.brand_dark,
    brandOn: c.brand_on,
    onBrand: c.on_brand,
    brand2: c.brand_2,
    brand2On: c.brand_2_on,
    gradients: { brand: c.gradient_brand, screen: c.gradient_screen },
  };
}

export const fetchBranding = () => api<BrandingResponse>('/branding');

/** Брендинг целиком: тема, название организации и герб. */
export function useBranding() {
  const status = session((st) => st.status);
  return useQuery({
    queryKey: ['branding'],
    queryFn: fetchBranding,
    enabled: status === 'active',
    staleTime: 5 * 60_000,
  });
}

/** Абсолютный адрес герба: сервер отдаёт путь, а не URL. */
export function logoUri(branding?: BrandingResponse | null): string | undefined {
  return branding?.logo_url ? `${API_URL}${branding.logo_url}` : undefined;
}
